### PR TITLE
poptracker: add XDG desktop file and format with nixfmt-rfc-style

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes
Adds a desktop item for poptracker, since it can be launched without any options, and it is useful for people who need to access it frequently. Also formats the package file with nixfmt-rfc-style since there were a few things that it was complaining about.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/po/poptracker/package.nix [hunk 1]
```diff
@@ -53,10 +57,28 @@ stdenv.mkDerivation (finalAttrs: {
...
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = "poptracker";
+      desktopName = "PopTracker";
+      exec = "poptracker";
+      comment = "Universal, scriptable randomizer tracking solution";
+      categories = [
+        "Game"
+        "Utility"
```
Comment (eclairevoyant) [inline pkgs/by-name/po/poptracker/package.nix:77]: any icon we can use?

Comment (pyrox0) [reply pkgs/by-name/po/poptracker/package.nix:77]: Is there a way to reference the `$out` directory from the makeDesktopItem call? Poptracker does include an icon in its assets directory(installed as `$out/share/poptracker`), but I couldn't find a way to reference it from the makeDesktopItem call.

Comment (eclairevoyant) [reply pkgs/by-name/po/poptracker/package.nix:77]: If it's installed into the correct dir (`$out/share/pixmaps/` or `$out/share/icons/hicolor/<size>/apps/`), simply the name of the icon should suffice.

Comment (pyrox0) [reply pkgs/by-name/po/poptracker/package.nix:81]: Added an icon

File: pkgs/by-name/po/poptracker/package.nix [hunk 2]
```diff
@@ -26,14 +29,15 @@ stdenv.mkDerivation (finalAttrs: {
   patches = [ ./assets-path.diff ];
 
   postPatch = ''
-     substituteInPlace src/poptracker.cpp --replace "@assets@" "$out/share/$pname/"
+    substituteInPlace src/poptracker.cpp --replace "@assets@" "$out/share/$pname/"
```
Comment (eclairevoyant) [inline pkgs/by-name/po/poptracker/package.nix:32]: We should be consistent, either `${finalAttrs.pname}` or `$pname` or string literal.
I don't see much benefit from using `finalAttrs` here personally, I'd rather just use the literal...

Comment (drupol) [reply pkgs/by-name/po/poptracker/package.nix:32]: Yes, `pname` shouldn't be reused somewhere else actually. Please use a string literal.
```suggestion
    substituteInPlace src/poptracker.cpp --replace "@assets@" "$out/share/poptracker/"
```

**Review by drupol — DISMISSED** [review]

File: pkgs/by-name/po/poptracker/package.nix (hunk 2 above)
Comment (pyrox0) [reply pkgs/by-name/po/poptracker/package.nix:32]: Fixed

**Comment by drupol** [comment]
Thanks!

Do you think adding the commit that reformat the file in this file would be possible too: https://github.com/NixOS/nixpkgs/blob/master/.git-blame-ignore-revs

**Review by freyacodes — APPROVED** [review]

File: pkgs/by-name/po/poptracker/package.nix [hunk 3]
```diff
@@ -60,9 +63,26 @@ stdenv.mkDerivation (finalAttrs: {
         libsForQt5.kdialog
       ]
     }
+    mkdir -p $out/share/icons/hicolor/{64x64,512x512}/apps
+    ln -s $out/share/${finalAttrs.pname}/icon.png  $out/share/icons/hicolor/64x64/apps/poptracker.png
+    ln -s $out/share/${finalAttrs.pname}/icon512.png  $out/share/icons/hicolor/512x512/apps/poptracker.png
```
Comment (SuperSandro2000) [inline pkgs/by-name/po/poptracker/package.nix:68]:
```suggestion
    ln -s $out/share/poptracker/icon.png  $out/share/icons/hicolor/64x64/apps/poptracker.png
    ln -s $out/share/poptracker/icon512.png  $out/share/icons/hicolor/512x512/apps/poptracker.png
```
there is a tracking issue about oversuing pname

File: pkgs/by-name/po/poptracker/package.nix [hunk 4]
```diff
@@ -53,10 +57,32 @@ stdenv.mkDerivation (finalAttrs: {
     runHook preInstall
     install -m555 -Dt $out/bin build/linux-x86_64/poptracker
     install -m444 -Dt $out/share/${finalAttrs.pname} assets/*
```
Comment (eclairevoyant) [inline pkgs/by-name/po/poptracker/package.nix:59]:
```suggestion
    install -m444 -Dt $out/share/poptracker assets/*
```
This line as well

**Pull request merged**